src/mpid/ch3/util/sock/ch3u_getinterfaces.c

```diff
@@ -199,9 +199,12 @@ static int MPIDI_CH3U_GetIPInterface( MPIDI_CH3I_Sock_ifaddr_t *ifaddr, int *fou
 	rc = MPL_env2bool( "MPICH_DBG_IFNAME", &dbg_ifname );
 	if (rc != 1) dbg_ifname = 0;
     }
-
+#if defined(AF_INET6)
```

Comment:
Did you mean `!defined`?

Reply:
Similar to the tcp netmod, the code should try to open a IPv4 socket and bind to it before trying IPv6.